src/databricks/labs/ucx/assessment/azure.py

```diff
@@ -247,3 +275,113 @@ def snapshot(self) -> Iterable[AzureServicePrincipalInfo]:
     def _try_fetch(self) -> Iterable[AzureServicePrincipalInfo]:
         for row in self._fetch(f"SELECT * FROM {self._schema}.{self._table}"):
             yield AzureServicePrincipalInfo(*row)
+
+
+@credentials_provider("azure-cli", ["host"])
+def _credentials(cfg: Config):
+    token_source = AzureCliTokenSource(cfg.arm_environment.service_management_endpoint)
+
+    def inner() -> dict[str, str]:
+        token = token_source.token()
+        return {"Authorization": f"{token.token_type} {token.access_token}"}
+
+    return inner
+
+
+class AzureResourcePermissions(CrawlerBase[AzureStorageSpnPermissionMapping]):
+    def __init__(self, ws: WorkspaceClient, lc: ExternalLocations, sql_backend: SqlBackend, inventory_schema: str):
+        super().__init__(
+            sql_backend, "hive_metastore", inventory_schema, "azure_storage_accounts", AzureStorageSpnPermissionMapping
+        )
+        self._rm_host = ws.config.arm_environment.resource_manager_endpoint
```

Comment:
Please separate the class that calls azure API from class that compares data with external locations. And probably we want to the mapping we get from azure api into csv, like we do for table mapping